server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -932,25 +929,21 @@ public void run() {
     final ServerContext context = getContext();
     final String zroot = getZooKeeperRoot();
 
-    this.compactionJobQueues = new CompactionJobQueues(
-        getConfiguration().getCount(Property.MANAGER_COMPACTION_SERVICE_PRIORITY_QUEUE_SIZE));
-
     // ACCUMULO-4424 Put up the Thrift servers before getting the lock as a sign of process health
     // when a hot-standby
     //
     // Start the Manager's Fate Service
     fateServiceHandler = new FateServiceHandler(this);
     managerClientHandler = new ManagerClientServiceHandler(this);
-    compactionCoordinator =
-        new CompactionCoordinator(context, tserverSet, security, compactionJobQueues, nextEvent);
+    compactionCoordinator = new CompactionCoordinatorImpl(context, tserverSet, security, nextEvent);
```

Comment:
This doesn't need to be done now but looking at this I'm thinking it might be nice if we also get rid of the Manager having to instantiate the Compaction coordinator impl directly with the constructor as it still has to know about the specific impl. If the goal here is to decouple everything into services than maybe we should go ahead and use `ServiceLoader` and make things like `CompactionCoordinator` pluggable in some way. The main use case I'm thinking of is to make it easier to test as in unit tests you could easily swap out for a dummy implementation but it would also allow plugging in other `CompactionCoordinator` impls in the future. 

The main challenge with using something like `ServiceLoader` is that obviously in this case `CompactionCoordinatorImpl() `is taking 4 arguments. So we could define a factory class API instead. This could look something like:

```java
// possible interface, this could also be more generic and just take an arbitary list of args
interface CompactionCoordinatorFactory {
 CompactionCoordinator create(ServerContext ctx, LiveTServerSet tservers,
                    SecurityOperation security, EventCoordinator eventCoordinator);
}

// sample default impl
class CompactionCoordinatorFactorImpl implements CompactionCoordinatorFactory {
  public CompactionCoordinator create(ServerContext ctx, LiveTServerSet tservers,
                             SecurityOperation security, EventCoordinator eventCoordinator) {
    return new CompactionCoordinatorImpl(ctx, tserverSet, security, nextEvent);
  }
}

// Inside Manager where we create the impl in the run() method
ServiceLoader<CompactionCoordinatorFactory> factory = ServiceLoader.load(CompactionCoordinatorFactory.class);
compactionCoordinator = factory.findFirst().orElseThrow().create(context, tserverSet, security, nextEvent);
```

Reply:
>  If the goal here is to decouple everything into services than maybe we should go ahead and use ServiceLoader and make things like CompactionCoordinator pluggable in some way. 

I did not have a goal of being able to swap out the impls at runtime.  I was more trying to wrangle the code into something that is easier to maintain.  The manager has distinct functionality, but some of the manager code is not well aligned with the functionality.  I thought the interfaces would help re align the code with the functionality by making dependencies between these functional service in the manager explicit in the code via using interfaces.  

As a secondary bonus, I thought this reorg may help make the code more testable.  However for the testing I was thinking of doing it would still be compile time switching of impls and not run time.